modules/cookbooks/pages/rag.adoc

```diff
@@ -0,0 +1,174 @@
+= Retrival Augmented Generation (RAG)
+// tag::single-source[]
+:description: How to configure Redpanda Connect to create a RAG pipeline using Postgres and PGVector.
+
+This cookbook demonstrates how to create a Retrival Augmented Generation (RAG) pipeline using Postgres and https://github.com/pgvector/pgvector[PGVector]. We'll create two Redpanda Connect pipelines: one for creating embeddings with https://ollama.ai[Ollama^] and another for searching the data using https://github.com/pgvector/pgvector[PGVector]. We will run both pipelines together using xref:guides/streams_mode/about.adoc[streams mode].
+
+== Compute the embeddings
+
+We'll start out by defining the indexing pipeline that will take textual data from Redpanda, compute embeddings for it, and then write it into a Postgres table with a PGVector index on the embeddings column.
+
+Articles to be indexed are received over the topic `articles` and look like this:
+
+[source,json]
+----
+{
```

Comment:
Do we need to talk about where this data is created?

Reply:
I think, that again assumes an experienced user. So if you want to lower the barrier, you could add it in?

Reply:
Added it in